✨login&register Ui

Echaimi opened the pull request:
# Description

Closes #(issue) | No issue linked.

## Type of change

Please add any relevant information to the type of change you've made.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement to something existing
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (or is a documentation update)
- [ ] HIGH IMPORTANCE (requires a fast merging)

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. It is way more likely for a PR to be merged and approved with successful tests.

- [x] Yes, I ran tests
- [ ] No, I did not include any test
- [ ] No test is needed for this change

Echaimi committed:
✨lgoin&register Ui

YummYume commented on compose.yml:
```diff
@@ -1,3 +1,5 @@
+version: '3.8'
```
```suggestion
```

C'est déprécié

YummYume commented on compose.yml:
```diff
@@ -12,6 +14,8 @@ services:
     tty: true
     restart: unless-stopped
     container_name: hk-app
+    environment:
+      - DATABASE_URL=${DATABASE_URL}
```
```suggestion
```

On utilise le `.env` en dev

YummYume commented on docker/dev/Dockerfile:
```diff
@@ -2,14 +2,24 @@ FROM oven/bun:debian
 
-# Switch to 'bun' user
-USER bun
+COPY package*.json ./
```
```suggestion
```

On a déjà un volume, pas besoin de COPY ni d'installer ici

YummYume commented on docker/dev/Dockerfile:
```diff
@@ -2,14 +2,24 @@ FROM oven/bun:debian
-USER bun
+COPY package*.json ./
+
+RUN npm install --legacy-peer-deps
```
```suggestion
```

On utilise Bun

YummYume commented on docker/dev/Dockerfile:
```diff
@@ -2,14 +2,24 @@ FROM oven/bun:debian
+
+RUN npm install --legacy-peer-deps
+
+COPY . .
+
+RUN chmod +x ./scripts/bun.sh
+
+RUN chown -R 1000:1000 /home/bun/app
+RUN chmod -R 775 /home/bun/app
+
+EXPOSE 5173
+EXPOSE 4173
```
```suggestion
```

Les permissions sont déjà bonnes avec le volume, et le expose est fait par le compose

YummYume commented on package.json:
```diff
@@ -55,7 +55,7 @@
     "tailwindcss": "^3.4.4",
     "tslib": "^2.6.3",
-    "typescript": "^5.5.2",
+    "typescript": "^5.4.3",
```
Remettre la bonne version

LeoBruant commented on compose.yml:
```diff
@@ -1,3 +1,5 @@
+version: '3.8'
+
```
Faut plus mettre la version avec la dernière version de compose

Echaimi replied:
je l'ai déja fix sur la prochaine pr mais plz stop review une ui et un patch pour vos migra on perd du temps pour

YummYume commented on src/routes/login/+page.svelte:
```diff
@@ -0,0 +1,10 @@
+<script lang="ts">
+    import LoginForm from './LoginForm.svelte';
+
+    const handleLogin = (event: CustomEvent<{ email: string; password: string }>) => {
+        const { email, password } = event.detail;
+        console.log('Login details:', email, password);
+    };
+</script>
```
On ne fait pas de `onsubmit`, etc... Car on utilise du [progressive enhancement](https://kit.svelte.dev/docs/form-actions#progressive-enhancement). Tu peux utiliser [Superforms](https://superforms.rocks/) pour faire le formulaire

YummYume commented on src/routes/login/LoginForm.svelte:
```diff
@@ -0,0 +1,102 @@
+                <div class="text-center text-sm">
+                    <p>
+                        Don't have an account?
+                        <button type="button" on:click="{navigateToSignup}" class="link-primary">
+                            Sign up
+                        </button>
```
Mettre un lien pas un bouton

LeoBruant replied:
Avec le composant Button de shadcn ca devient un lien quand on met un href

YummYume commented on src/routes/login/LoginForm.svelte:
```diff
@@ -0,0 +1,102 @@
+        <img
+            src="/medical-background.jpg"
+            alt="Medical Background"
+            class="h-full w-full object-cover"
```
```suggestion
            class="h-full w-full object-cover"
            loading="lazy"
```

YummYume commented on src/routes/login/LoginForm.svelte:
```diff
@@ -0,0 +1,102 @@
+</div>
+
+<style>
+    .half-width {
```
Utiliser Tailwind `w-2/4`

LeoBruant replied:
`w-1/2`

YummYume commented on src/routes/login/LoginForm.svelte:
```diff
@@ -0,0 +1,102 @@
+                <input type="hidden" name="remember" value="true" />
+                <div class="-space-y-px rounded-md shadow-sm">
+                    <div>
+                        <label for="email-address" class="sr-only">Email address</label>
```
Le site est en français

YummYume commented on src/routes/login/LoginForm.svelte:
```diff
@@ -0,0 +1,102 @@
+<script lang="ts">
+    import { goto } from '$app/navigation';
+
+    let email = '';
+    let password = '';
+
+    const navigateToSignup = () => {
+        goto('/signup');
+    };
+</script>
```
```suggestion
```

YummYume commented on src/routes/signup/+page.svelte:
```diff
@@ -0,0 +1,12 @@
+<script lang="ts">
+    import SignupForm from './SignupForm.svelte';
+
+    const handleSignup = (
```
Pareil que pour le login

LeoBruant commented on src/app.scss:
```diff
@@ -52,8 +60,8 @@
 
-    --primary: 23 100% 83%;
-    --primary-foreground: 222.2 47.4% 11.2%;
+    --primary: 15 97% 53%;
+    --primary-foreground: 210 40% 98%;
```
Faut laisser les couleurs du dark mode

LeoBruant commented on src/app.scss:
```diff
@@ -5,8 +5,12 @@
+    --background-secondary: 0 0% 95%;
     --foreground: 222.2 84% 4.9%;
 
+    --form-container-bg: 0 0% 98%;
+    --form-container-bg-dark: 222.2 84% 12%;
+
```
Vaut mieux laisser les variables de base de shadcn https://www.shadcn-svelte.com/docs/theming

YummYume commented on src/routes/signup/SignupForm.svelte:
```diff
@@ -0,0 +1,143 @@
+</div>
+
+<style>
+    .form-container {
+        flex: 1;
+        display: flex;
+        justify-content: center;
+        align-items: center;
+    }
+
+    .placeholder {
+        flex: 1;
+        background-color: var(--muted);
+        display: flex;
+        align-items: center;
+        justify-content: center;
+    }
+
+    .space-y-8 > * + * {
+        margin-top: 2rem;
+    }
```
Mettre dans le .scss ou juste utiliser Tailwind pour respecter le style

YummYume commented on vite.config.ts:
```diff
@@ -10,4 +10,12 @@ export default defineConfig({
       autoInstall: true,
     }),
   ],
+  server: {
+    host: '0.0.0.0',
+    port: 5173,
+    strictPort: true,
+    hmr: {
+      host: 'localhost',
+    },
+  },
```
Je pense pas qu'il y ai besoin (déjà par défaut)

Echaimi committed:
Update compose.yml

Co-authored-by: Yam <61044908+YummYume@users.noreply.github.com>

Co-authored-by: Yam <61044908+YummYume@users.noreply.github.com>

Co-authored-by: Yam <61044908+YummYume@users.noreply.github.com>

Update docker/dev/Dockerfile

Update src/routes/login/LoginForm.svelte

Co-authored-by: Yam <61044908+YummYume@users.noreply.github.com>

fix review

fix reviex

review

LeoBruant commented:
Attention la CI

Echaimi committed:
review

Echaimi commented:
> Attention la CI
nan jure
